vars/yamlSubstituteVariables.groovy

```diff
@@ -0,0 +1,313 @@
+import com.sap.piper.ConfigurationHelper
+import com.sap.piper.GenerateDocumentation
+import com.sap.piper.variablesubstitution.ExecutionContext
+import com.sap.piper.variablesubstitution.Logger
+import groovy.transform.Field
+
+import static com.sap.piper.Prerequisites.checkScript
+
+@Field Logger logger = new Logger()
+@Field String STEP_NAME = getClass().getName()
+@Field Set GENERAL_CONFIG_KEYS = []
+@Field Set STEP_CONFIG_KEYS = GENERAL_CONFIG_KEYS + [
+    /**
+     * The {@code String} path of the Yaml file to replace variables in. <br>
+     * Defaults to "manifest.yml" if not specified otherwise.
+     */
+    'manifestFile',
+    /**
+     * The {@code String} path of the Yaml file to produce as output.
+     * If not specified this will default to {@code manifestFile} and overwrite it.
+     */
+    'outputManifestFile',
+    /**
+     * The {@code String} path of the Yaml file containing the variables' values to use as a replace in the manifest file. <br>
+     * Defaults to "manifest-variables.yml" if not specified otherwise
+     */
+    'variablesFile'
+]
+
+@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS
+
+/**
+ * Step that substitutes variables in a given YAML file with those specified in a another. The format to reference a variable
+ * in the YAML file is to use double parentheses {@code ((} and {@code ))}, e.g. {@code ((variableName))}.<br>
+ * A declaration of a variable and assignment of its value is simply done as a property in the variables YAML file.
+ * <p>
+ * The format follows <a href="https://docs.cloudfoundry.org/devguide/deploy-apps/manifest-attributes.html#variable-substitution">Cloud Foundry standards</a>.
+ * <p>
+ * The step is activated by the presence of both a {@code manifest.yml} and a variables file. Names of both files are configurable.
```

Comment:
Applies here and elsewhere: The javadoc syntax `@code` and similar does not work here. We are using some home made docu-generator only providing a very limited set of features similar to the well know javadoc toolset. What we use is basically some pre-preprocessing for `mkdocs`. In the docu we have to use `mkdocs` syntax rather than `javadoc` syntax. That is weird but that is the way it is. Basically there is no suitable javadoc generator for our use case available (e.g. the signature to `call` is always a map, but the 'real' signature are the key/value pairs inside the map.

Reply:
Ok! Good to know. I saw `{@code ...}` in other places of the lib so I assumed it would work.
So how (in `mkdocs` syntax) would you demarcate a term that is supposed to be highlighted like a `{@code ...}` block?
I scanned the `mkdocs` syntax but could not find something quickly - only fenced code blocks, which is not what is needed here.
Do you suggest to remove the `{@code }` blocks altogether?

Reply:
For the lack of better best practice guidance, I replaced this with simply markdown used to demarcate code, i.e. single ticks.